l10n_it_fatturapa_in/wizard/wizard_import_fatturapa.py

```diff
@@ -1163,10 +1163,81 @@ def set_welfares_fund(
             return
 
         WelfareFundLineModel = self.env['welfare.fund.data.line']
+
+        wts = False
+        enasarco_relax_checks = self.env.user.company_id.enasarco_relax_checks
+        if any([welfareLine for welfareLine in Welfares
+                if welfareLine.TipoCassa == 'TC07']):
+            # Search the matching withholding tax
+            wts = self.env['withholding.tax'].search(
+                [('wt_types', '=', 'enasarco')])
+
+        processedLines = []
         for welfareLine in Welfares:
-            WalferLineVals = self._prepareWelfareLine(
-                invoice.id, welfareLine)
+            # Handle only TC07
+            if welfareLine.TipoCassa == 'TC07':
+                if not wts:
+                    msg = _(
+                        "The bill contains Welfare Fund tax with "
+                        "Type %s, "
+                        "but such a tax is not found in your system. Please "
+                        "set it."
+                    ) % welfareLine.TipoCassa
+                    if enasarco_relax_checks:
+                        self.log_inconsistency(msg)
+                    else:
+                        raise UserError(msg)
+
+                wt_found = False
+                for wt in wts:
+                    if wt.tax == float(welfareLine.AlCassa):
+                        wt_found = wt
+                        break
+
+                if not wt_found:
+                    msg = _(
+                        "The bill contains Welfare Fund tax with "
+                        "Type %s and Tax %s "
+                        "but such a tax is not found in your system. Please "
+                        "set it."
+                    ) % (welfareLine.TipoCassa, float(welfareLine.AlCassa))
+                    if enasarco_relax_checks:
+                        self.log_inconsistency(msg)
+                    else:
+                        raise UserError(msg)
+
+            WalferLineVals = self._prepareWelfareLine(invoice.id, welfareLine)
             WelfareFundLineModel.create(WalferLineVals)
+
+            # We are not interested to continue if Enasarco is used
+            if welfareLine.TipoCassa == 'TC07':
+                found = False
+                for line in invoice.invoice_line_ids:
+                    # search an invoice line having the same amount of the
+                    # ImportoContributoCassa
```

Comment:
Non capisco il commento: se non sbaglio non stai cercando righe in base all'importo

Reply:
Era il vecchio, lo tolgo! Tanto questa parte poi la dobbiamo discutere perchè ho l'impressione che ci saranno dei casi particolari che dipende da come fatturano gli agenti. Poi ne parliamo.